### ISSUE TITLE
require.config.js is not built properly.

### ISSUE THREAD

**Issue by Kaik** [zikula/core#2914]
| Q                 | A
| ----------------- | ---
| Zikula Version    | 1.4.x
| PHP Version       | 5.6

### Expected behavior
Proper baseUrl to components in require.config.js

### Actual behavior
baseUrl is set to "components" (default) 

### Steps to reproduce
Install Zikula and try to use require.js with require.config.js
```
    <script src="{{ asset('require.js') }}"></script>
    <script src="{{ asset('require.config.js') }}"></script> 
    <script>
        require(['jquery'
        ], function ($) {
        }
        );
    </script>  
```

I have tried to use default require.js config created by composer-installers which is present in /web folder but baseUrl to components is wrong this is mostly because it is not configured in our composer.json or should be configured in different place more information's here https://github.com/RobLoach/component-installer#base-url
Changing baseUrl simple to web does not solve this problem as it uses "current url" as part before /web so "http://www.domain.com/en/whatever/web" the only way it works is to use ``` Zikula.Config.baseURL + "web" ```

The second attempt was to use require-buld.js but something was wrong when using installed Backbone

I have ended up using custom require config inside my module main.js file
```
// Require.js allows us to configure shortcut alias
require.config({
    // The shim config allows us to configure dependencies for
    // scripts that do not call define() to register a module
    shim: {
        underscore: {
            exports: '_'
        },
        backbone: {
            deps: [
                'underscore',
                'jquery'
            ],
            exports: 'Backbone'
        }
    },
    paths: {
        jquery: Zikula.Config.baseURL + 'web/jquery/jquery',
        underscore: Zikula.Config.baseURL + 'web/underscore/underscore',
        backbone: Zikula.Config.baseURL + 'web/backbone/backbone'
    }
});
```
I will try to investigate it further. I'm leaving this issue here for now maybe you have idea how can we solve this.

**Comment by craigh** [comment]
> Install Zikula and try to use require.js with require.config.js

are these libraries currently part of the core?

**Comment by Kaik** [comment]
This is installed by roblach component installer and configured automatically.

**Comment by Kaik** [comment]
Best way would be to use ``` Zikula.Config.baseURL + "web" ``` as baseUrl but I do not know if this is possible to do in composer.json. The other way might be to override somehow, somewhere https://github.com/RobLoach/component-installer/blob/master/src/ComponentInstaller/Process/RequireJsProcess.php

**Comment by Guite** [comment]
assigning this to @Kaik as you know the topic :)

**Comment by craigh** [comment]
> Best way would be to use `Zikula.Config.baseURL + "web"` as baseUrl but I do not know if this is possible to do in composer.json.

`Zikula.Config.baseURL + "web"` is not available in composer since it is generated in html, right?

**Comment by Guite** [comment]
The solution can't be in composer. The CI server for example has another path than the user using the created package later on.

**Comment by Kaik** [comment]
>since it is generated in html, right?

Do not understand, ``` Zikula.Config.baseURL ``` does not have to be available in the composer. It is used later on in require.config.js where it is available. The problem is that there might be no way to express it in a way that it will be a valid javascript. As at the end it is using javascript to create proper baseUrl.

I have looked into automatic creation of this file and this might be impossible to do without overriding this https://github.com/RobLoach/component-installer/blob/master/src/ComponentInstaller/Process/RequireJsProcess.php#L192 it basically creates json and doubt there is a way to add working js there :(

**Comment by Guite** [comment]
But when overriding this class you can of course use PHP instead...

**Comment by craigh** [comment]
I'd prefer to not use `Zikula.Config.baseURL` anyway since it requires legacy (Scriptaculous based) `Zikula.js`. It would be better to use Twig and get the same information from `app.request`

**Comment by Kaik** [comment]
Well, this has nothing to do with Twig at this point just plain PHP and js. Second, we might want to have Zikula.js we may need to rewrite it but this kind of object is needed.

**Comment by Kaik** [comment]
The way to accomplish what we want is to add ```components.baseUrl = Zikula.Config.baseURL + "web";``` below this line https://github.com/RobLoach/component-installer/blob/master/src/ComponentInstaller/Process/RequireJsProcess.php#L199
But overriding the whole thing just to add one line is kind of bad. I wonder if simple post-install and post-update script could not read this file and add what we need where we need.

**Comment by Guite** [comment]
> I wonder if simple post-install and post-update script could not read this file and add what we need where we need. 
probably

**Comment by Guite** [comment]
Fixed in #2995

**Comment by craigh** [comment]
nice work @Kaik ! thanks!
